test/topology_custom/test_tablets2.py

```diff
@@ -1521,6 +1521,59 @@ async def assert_empty_table():
     logger.info("Verify data is not resurrected")
     await assert_empty_table()
 
+
+# Reproducer for: https://github.com/scylladb/scylla-enterprise/issues/5100
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_drop_keyspace_while_split(manager: ManagerClient):
+
+    # This tests if the split ready compaction groups are correctly created
+    # on a shard with several storage groups for the same table
+
+    logger.info("Bootstrapping cluster")
+    cmdline = [ '--target-tablet-size-in-bytes', '8192',
+                '--smp', '2' ]
+    config = { 'error_injections_at_startup': ['short_tablet_stats_refresh_interval'] }
+    servers = [await manager.server_add(config=config, cmdline=cmdline)]
+
+    s0_log = await manager.server_open_log(servers[0].server_id)
+
+    cql = manager.get_cql()
+    await wait_for_cql_and_get_hosts(cql, [servers[0]], time.time() + 60)
+
+    await manager.api.disable_tablet_balancing(servers[0].ip_addr)
+
+    # create a table so that it has at least 2 tablets (and storage groups) per shard
+    await cql.run_async("CREATE KEYSPACE test WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} AND tablets = {'initial': 4};")
```

Comment:
nit: we are trying to move away from test.test and use unique keyspace name (see https://github.com/scylladb/scylladb/pull/22334 )

Reply:
https://github.com/scylladb/scylladb/pull/22399 is more comprehensive and will override #22334